### ISSUE TITLE
[MODULES-250] ModuleClassLoader skips bootstrap class loader for custom namespaces

### ISSUE THREAD

**Issue by dmlloyd** [jboss-modules/jboss-modules#678]
**[Rafael Winterhalter](https://issues.redhat.com/secure/ViewProfile.jspa?name=raphw)** opened **[MODULES-250](https://issues.redhat.com/browse/MODULES-250?redirect=false)** and commented

The JBoss module class loader (http://docs.jboss.org/osgi/jboss-osgi-1.0.0/apidocs/org/jboss/modules/ModuleClassLoader.html) does not load classes on the bootstrap class loader if they are not within the java.* namespace. This makes it impossible to inject custom classes into the bootstrap class loader what is customary for some tools and Javaagents.

It also violates the class loader contract.

---
No further details from [MODULES-250](https://issues.redhat.com/browse/MODULES-250?redirect=false)

**Comment by dmlloyd** [comment]
@dmlloyd commented

If you want to add additional packages to all modules, use the `jboss.modules.system.pkgs` system property.  If you're running in the context of WildFly, check out the `standalone.conf` file for more information.

**Comment by dmlloyd** [comment]
**[Rafael Winterhalter](https://issues.redhat.com/secure/ViewProfile.jspa?name=raphw)** commented

Runtime-agnostic tooling might not be aware that it is run on JBoss but is still allowed to append classes to the boostrap loader using the instrumentation API where it then expects to load these classes from any class loader within the JVM in question. For all I know, not attempting to load these classes violates the class loader's implementation contract. The JBoss class loader is the only one out of dozens that we have looked at that suffered from this.

The work-around is to put these classes within a Java package so we have solved this on our side but for compatibility with other tools I recommend you to not make this assumption.
